Core/include/Acts/Surfaces/detail/VerticesHelper.hpp

```diff
@@ -62,6 +62,12 @@ void createSegment(std::vector<vertex_t>& vertices,
     vertex_t vertex = vertex_t::Zero();
     vertex(0) = rxy.first * std::cos(phi);
     vertex(1) = rxy.second * std::sin(phi);
+
+    if(angle != 0. && vertex.size() == 3) {
+      double sign = offset(2) != 0. ? std::fabs(offset(2))/offset(2) : 1.;
+      vertex(2) = vertex(2) + sign * vertex(1) * std::tan(angle);
+    }
+
```

Comment:
Is this actually called anywhere in the current changeset?

What I was originally thinking (and please correct me if I'm missing something), was to move this part:

https://github.com/acts-project/acts/blob/98597a7191e142015964c6dba5a3f535b6ace905/Core/src/Surfaces/CylinderSurface.cpp

into the bounds itself. Basically: `CylinderSurface` would call on the bounds to deliver (or maybe fill via `push_back`) the `vertices` vector. 

The reason for this is that in principle, `CylinderSurface` handles the coordinate system, and doesn't really care about the specific bounds that have been set.

With this approach, we could confine the fraction of the code that needs to be aware of this `angle` parameter, and also we would have the option of using a separate class for this.

Reply:
Thanks for the feedback, this is exactly why I made the draft to get comments whether I was doing something non-optimal. OK, will try this approach instead

Reply:
I messed up the code snippet linked in the comment above. Fixed now.

Reply:
An attempt has been made to move this to `CylinderBounds`